GameOfRunes/GameOfRunes/GameEngine/GameEngine.swift

```diff
@@ -16,9 +16,9 @@ class GameEngine: GameEngineFacade {
     private var entities = [EntityType: Set<Entity>]()
     private(set) var systems = [ComponentType: System]()
     private var toRemoveEntities = Set<Entity>()
-    private(set) var gameScene: GameSceneFacade?
-    var metadata: GameMetaData
-    
+    let metadata: GameMetaData
+    weak var rootRenderNode: RootRenderNode?
```

Comment:
we should pass the root node in instead of setting it from outside

Reply:
Yup I can refactor into this, but the thing is that the PhysicsWorld's contactDelegate still has to be set from GameScene. Since the gameEngine inside RootRenderNode is of type `GameEngineFacade`, maybe just leave this outside of RootRenderNode so we can avoid the ugly typecasting? This is just a slight concern hahahaha

Reply:
But there is a circular thing going on. GameEngine needs to have RootRenderNode. RootRenderNode needs to have GameEngine.

Reply:
pass in self to `gameEngine's init`, then we don't need to expose `rootRenderNode`

Reply:
But self is not defined before the super.init() call, and the code creates Game Engine before the super.init() call because gameEngine is a parameter. 

I can make gameEngine optional, and create it after the super.init() call, i guess?

Reply:
Resolved!